gramps_webapi/api/resources/exporters.py

```diff
@@ -0,0 +1,372 @@
+    options.current_year = args["current_year"]
+    options.years_after_death = args["years_after_death"]
+    options.reference = args["reference"]
+    options.include_individuals = int(args["include_individuals"])
+    options.include_marriages = int(args["include_marriages"])
+    options.include_children = int(args["include_children"])
+    options.include_places = int(args["include_places"])
+    options.include_witnesses = int(args["include_witnesses"])
+    options.include_media = int(args["include_media"])
+    options.translate_headers = int(args["translate_headers"])
+    options.compression = int(args["compress"])
+    if args["person"] is None:
+        if args["gramps_id"] is not None or args["handle"] is not None:
+            abort(422)
+    else:
+        if args["gramps_id"] is not None:
+            gramps_id = args["gramps_id"]
+            if db_handle.get_person_from_gramps_id(gramps_id) is None:
+                abort(422)
+        else:
+            try:
+                person = db_handle.get_person_from_handle(args["handle"])
+            except HandleError:
+                abort(422)
+            gramps_id = person.gramps_id
+        try:
+            options.set_person_filter(args["person"], gramps_id)
+        except ValueError:
+            abort(422)
+    if args["event"] is not None:
+        try:
+            options.set_event_filter(args["event"])
+        except ValueError:
+            abort(422)
+    if args["note"] is not None:
+        try:
+            options.set_note_filter(args["note"])
+        except ValueError:
+            abort(422)
+    try:
+        options.set_proxy_order(args["sequence"])
+    except ValueError:
+        abort(422)
+    if args["locale"] is not None:
+        options.locale = get_locale_for_language(args["locale"])
+        if options.locale is None:
+            abort(422)
+    return options
+
+
+def run_export(db_handle: DbReadBase, extension: str, options):
+    """Generate the export."""
+    export_path = TEMP_DIR
+    if current_app.config.get("EXPORT_PATH"):
+        export_path = current_app.config.get("EXPORT_PATH")
+    file_name = os.path.join(export_path, "{}.{}".format(uuid.uuid4(), extension))
+    _resources = ResourcePath()
+    os.environ["GRAMPS_RESOURCES"] = str(Path(_resources.data_dir).parent)
+    filters.reload_custom_filters()
+    plugin_manager = BasePluginManager.get_instance()
+    for plugin in plugin_manager.get_export_plugins():
+        if extension == plugin.get_extension():
+            export_function = plugin.get_export_function()
+            result = export_function(db_handle, file_name, User(), options)
+            if not result:
+                abort(500)
+            return file_name, "." + extension
+
+
+class ExportersResource(ProtectedResource, GrampsJSONEncoder):
+    """Exporters resource."""
+
+    @property
+    def db_handle(self) -> DbReadBase:
+        """Get the database instance."""
+        return get_dbstate().db
```

Comment:
Given this property is used by 3 classes, could it be turned into a function?

Reply:
Yeah I just sort of carried that all over the place from your use of it for some unknown reason, but yeah it is unnecessary as `get_dbstate().db` could be used directly.  I can factor that out.